pkgs/sdk/server/src/Internal/DataSystem/IDataSystem.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Threading.Tasks;
+using LaunchDarkly.Sdk.Server.Interfaces;
+using LaunchDarkly.Sdk.Server.Subsystems;
+
+namespace LaunchDarkly.Sdk.Server.Internal.DataSystem
+{
+    internal interface IReadOnlyStore
+    {
+        DataStoreTypes.ItemDescriptor? Get(DataStoreTypes.DataKind kind, string key);
+        DataStoreTypes.KeyedItems<DataStoreTypes.ItemDescriptor> GetAll(DataStoreTypes.DataKind kind);
+
+        bool Initialized();
+
+        DataStoreTypes.InitMetadata GetMetadata();
+    }
+
+    internal interface IFlagChanged
```

Comment:
This was only available on the concrete implementation of data source updates, so the flag tracker was dependent on the concrete implementation.